### PR TITLE
Remove type SubstituteId 

### PR DESCRIPTION
**Pull request by clintfred**
...and make string values of PolicyGrant components available to the outside world

### CODE REVIEW

File: src/policy.rs [hunk 1]
```diff
@@ -84,13 +84,13 @@ use std::convert::TryFrom;
 /// The triple (`category`, `sensitivity`, `data_subject`) maps to a single policy rule. Each policy
 /// rule may generate any number of users/groups.
 ///
-/// `substitute_user_id` replaces `%USER%` in a matched policy rule.
+/// `substitute_user_id replaces `%USER%` in a matched policy rule.
```
Comment (ernieturner) [inline src/policy.rs]: Did you mean to remove this backtick?

File: src/internal/user_api/mod.rs [hunk 2]
```diff
@@ -19,7 +19,7 @@ mod requests;
 #[derive(PartialEq, Debug, Serialize, Deserialize, Clone, Eq, Hash)]
 pub struct UserId(pub(crate) String);
 impl UserId {
-    pub fn id(&self) -> &String {
+    pub fn id(&self) -> &str {
```
Comment (ernieturner) [inline src/internal/user_api/mod.rs]: Is this considered a breaking change? This struct gets exposed externally, right?

File: src/policy.rs (hunk 1 above)
Comment (clintfred) [reply src/policy.rs]: good catch.

File: src/internal/user_api/mod.rs (hunk 2 above)
Comment (clintfred) [reply src/internal/user_api/mod.rs]: Yes it is part of our public API and is technically a breaking change. For most (all?) cases `&str` is more idiomatic than `&String` (similar to `&Option<T>` vs `Option<&T>`). If it's actually a breaking change depends on on the usage. Rust provides From and Deref impls that mitigate this. There was one change required to ironoxide-java (`.clone()` -> `.to_string()`)

We undoubtedly have more of these `&String` instances, and I could certainly see the sense in changing them all at once vs piecemeal.

**Pull request merged**